src/box/replication.cc

```diff
@@ -1030,6 +1031,8 @@ replicaset_is_connected(struct replicaset_connect_state *state,
 		struct applier *applier = appliers[i];
 		if (applier->state == APPLIER_CONNECTED) {
 			state->connected++;
+			if (!applier->ballot.is_booted)
+				state->booting++;
```

Comment:
A few lines above the counters `connected` and `failed` are nullified. Why don't you nullify `booting` there too?

Reply:
That's an oversight. Thanks for noticing!

Fixed:
```diff
diff --git a/src/box/replication.cc b/src/box/replication.cc
index 9b18510946..177c985baa 100644
--- a/src/box/replication.cc
+++ b/src/box/replication.cc
@@ -207,8 +207,8 @@ struct replicaset_connect_state {
        /** Number of successfully connected appliers. */
        int connected;
        /**
-        * Number of appliers which were connected to a loading instance. A
-        * subset of \a connected excluded from replicaset_sync_quorum
+        * Number of appliers which were connected to a loading instance. This
+        * is a subset of \a connected excluded from replicaset_sync_quorum
         * calculation.
         */
        int booting;
@@ -1026,6 +1026,7 @@ replicaset_is_connected(struct replicaset_connect_state *state,
        }
        /* Update connected and failed counters. */
        state->connected = 0;
+       state->booting = 0;
        state->failed = 0;
        for (int i = 0; i < count; i++) {
                struct applier *applier = appliers[i];
@@ -1064,6 +1065,8 @@ replicaset_connect(const struct uri_set *uris,
 {
        struct replicaset_connect_state state;
        memset(&state, 0, sizeof(state));
+       fiber_cond_create(&state.wakeup);
+
        if (uris->uri_count == 0) {
                replicaset_set_sync_quorum(&state);
                /* Cleanup the replica set. */
@@ -1121,8 +1124,6 @@ replicaset_connect(const struct uri_set *uris,
        /* Memory for on_state triggers registered in appliers */
        struct applier_on_connect triggers[VCLOCK_MAX];
 
-       fiber_cond_create(&state.wakeup);
-
        double timeout = replication_connect_timeout;
 
        /* Add triggers and start simulations connection to remote peers */
```